build/php/tests/Triniti/Tests/Schemas/Sys/RedirectIdTest.php

```diff
@@ -0,0 +1,96 @@
+<?php
+declare(strict_types=1);
+
+namespace Triniti\Tests\Schemas\Sys;
+
+use PHPUnit\Framework\TestCase;
+use Triniti\Schemas\Sys\RedirectId;
+use Gdbots\Pbj\Exception\InvalidArgumentException;
+
+final class RedirectIdTest extends TestCase
+{
+    public function testEmptyUri(): void
+    {
+        $this->expectException(InvalidArgumentException::class);
+
+        $uri = '';
+        RedirectId::fromUri($uri);
+    }
+
+    public function testFromUri(): void
+    {
+        $uri = '/unicode/fake/partial/url';
+        $id = RedirectId::fromUri($uri);
+        $this->assertInstanceOf('Triniti\Schemas\Sys\RedirectId', $id);
+        $this->assertSame(true, $this->isValid($id->toString()));
+
+        $uri = 'https://test.com/unicode/characters/✓ à la mode';
+        $id = RedirectId::fromUri($uri);
+        $this->assertInstanceOf('Triniti\Schemas\Sys\RedirectId', $id);
+        $this->assertSame(true, $this->isValid($id->toString()));
+
+        $uri = '/\n';
+        $id = RedirectId::fromUri($uri);
+        $this->assertInstanceOf('Triniti\Schemas\Sys\RedirectId', $id);
+        $this->assertSame(true, $this->isValid($id->toString()));
+
+        $uri = 'http:/test.com/test/url/here';
+        $id = RedirectId::fromUri($uri);
+        $this->assertInstanceOf('Triniti\Schemas\Sys\RedirectId', $id);
+        $this->assertSame(true, $this->isValid($id->toString()));
```

Comment:
@hunyoboy id toString is not the same as id->toUri.. you need to validate that scenario as well